lib/rubocop/cop/lint/literal_in_interpolation.rb

```diff
@@ -55,15 +58,22 @@ def offending?(node)
           node &&
             !special_keyword?(node) &&
             prints_as_self?(node) &&
-            # Special case for Layout/TrailingWhitespace
-            !(space_literal?(node) && ends_heredoc_line?(node))
+            # Special case for `Layout/TrailingWhitespace`
+            !(space_literal?(node) && ends_heredoc_line?(node)) &&
+            # Handled by `Lint/ArrayLiteralInRegexp`
+            !array_in_regexp?(node)
         end
 
         def special_keyword?(node)
           # handle strings like __FILE__
           (node.str_type? && !node.loc.respond_to?(:begin)) || node.source_range.is?('__LINE__')
         end
 
+        def array_in_regexp?(node)
+          grandparent = node.parent&.parent
+          node.array_type? && grandparent&.regexp_type?
```

Comment:
Is safe navigation necessary? If it is, adding a test would be appropriate.

```suggestion
          grandparent = node.parent.parent
          node.array_type? && grandparent.regexp_type?
```

Reply:
While running `bundle exec rake default` I was getting an error during rubocop inspection that `node.parent` was nil without safe navigation. However, I can no longer replicate this so I will remove the safe navigation.